src/codegen/compile.ml

```diff
@@ -1192,6 +1215,7 @@ module Stack = struct
     compile_unboxed_const n_bytes ^^
     G.i (Binary (Wasm.Values.I32 I32Op.Sub)) ^^
     set_stack_ptr env ^^
+    update_stack_min env ^^
```

Comment:
The assumption is that the allocated stack space is certainly accessed before the next allocation, in order to trap on an overflow (with allocation sizes < page size). Otherwise, multiple smaller allocations (without intermediate accesses) could skip below the last invalid page. I guess this is always guaranteed?

Reply:
Yeah, that was bothering me a little. There is no hard guarantee beyond assuming the code generator won't do stack allocation without using the space shortly. Maybe it would actually be better to always check, even when small or add an explicit argument to suppress the check so it's more obvious at call sites.

Reply:
What if we instead forced a read from the last word, only when the allocation is small, before returning?
That would be slower, but safer.
@luc-blaeser @ggreif?
I'll open a PR on master to measure the cost of this...

Reply:
See #3823